lib/archethic_web/graphql_schema/resolver.ex

```diff
@@ -212,6 +214,37 @@ defmodule ArchethicWeb.GraphQLSchema.Resolver do
     }
   end
 
+  def beacon_chain_summary(datetime) do
+    current_datetime = DateTime.utc_now()
+
+    next_datetime_summary_time =
+      datetime
+      |> BeaconChain.next_summary_date()
+
+    previous_current_date_summary_time =
+      current_datetime
+      |> BeaconChain.next_summary_date()
+
+    authorized_nodes = P2P.authorized_and_available_nodes()
+
+    cond do
+      DateTime.compare(next_datetime_summary_time, previous_current_date_summary_time) == :gt ->
+        {
+          :error,
+          "No data found at this date"
+        }
+
+      DateTime.compare(next_datetime_summary_time, previous_current_date_summary_time) == :eq ->
+        {
+          :ok,
+          BeaconChain.fetch_and_aggregate_summaries(datetime, authorized_nodes)
```

Comment:
Also there you should pass the `next_datetime_summary_time` value instead of `datetime` because if datetime is not an exact summary time it will return an error as the summary is not found

Reply:
in this case they are equal but I will change it for clarity.